app/models/user_track.rb

```diff
@@ -41,6 +41,30 @@ def self.for(user_param, track_param, external_if_missing: false)
     end
   end
 
+  def exercises
+    enabled_exercises
```

Comment:
I'd suggest memoizing this directly (and inlining `filter_enabled_exercises(track.exercises)`) rather than calling out to the other method.